### PR TITLE
feat(1010): [1] enabling prChain 2nd step

### PR DESCRIPTION
**Pull request by sakka2**
## Context
This PR is the second step of enabling prChain feature.
This time we fixed only a backend（api, models, workflow-graph）. UI will be fixed next time.

## Objective
- If `prChain` flag is true in screwdriver.yaml, run subsequent jobs following workflow graph.
- Create all jobs with `PR-${prNum}:` prefix.

## Other PRs
https://github.com/screwdriver-cd/screwdriver/pull/1444
https://github.com/screwdriver-cd/models/pull/315

## References
- New design of prChain feature
https://github.com/screwdriver-cd/screwdriver/issues/1010#issuecomment-448142968

### CODE REVIEW

File: lib/getNextJobs.js [hunk 1]
```diff
@@ -21,6 +21,9 @@ const getNextJobs = (workflowGraph, config) => {
         throw new Error('Must provide a PR number with "~pr" trigger');
     }
 
+    // Check if the job is triggerd by prChain with regexp
+    const prChainTrigger = config.trigger.match(/^(PR-[0-9]+:)?[\w-]+$/u);
```
Comment (tk3fftk) [inline lib/getNextJobs.js]: I think we should use a regexp of dataschema instead of defined here if we can.

File: lib/getNextJobs.js [hunk 2]
```diff
@@ -40,11 +43,17 @@ const getNextJobs = (workflowGraph, config) => {
...
         } else if (edge.src === config.trigger) {
             // Make PR jobs PR-$num:$cloneJob (not sure how to better handle multiple PR jobs)
-            if (config.trigger === '~pr' && !config.prChain) {
+            if (config.trigger === '~pr') {
                 jobs.add(`PR-${config.prNum}:${edge.dest}`);
             } else {
                 jobs.add(edge.dest);
             }
+        } else if (config.prChain) {
+            if (config.trigger === '~pr') {
```
Comment (tk3fftk) [inline lib/getNextJobs.js]: Is this if statement reachable?

File: lib/getNextJobs.js (hunk 1 above)
Comment (sakka2) [reply lib/getNextJobs.js]: Ok, I will use data-schema.

File: test/lib/getNextJobs.test.js [hunk 3]
```diff
@@ -36,7 +36,27 @@ describe('getNextJobs', () => {
             trigger: '~pr',
             prNum: '123',
             prChain: true
-        }), ['main']);
+        }),
+        [
+            'PR-123:main',
```
[the remark this answers is not in the document]
Comment (sakka2) [reply test/lib/getNextJobs.test.js]: Not, parallel. I will think about this again.

[the remark this answers is not in the document]
Comment (sakka2) [reply test/lib/getNextJobs.test.js]: I considered  this, and fixed to get chained jobs in models.
https://github.com/screwdriver-cd/models/pull/315/files#diff-679e90b5b15ab1b73ddb7ed2377e7413R375

File: lib/getNextJobs.js (hunk 1 above)
Comment (sakka2) [reply lib/getNextJobs.js]: I fixed to use data-schema

File: lib/getNextJobs.js [hunk 4]
```diff
@@ -40,11 +45,13 @@ const getNextJobs = (workflowGraph, config) => {
             }
         } else if (edge.src === config.trigger) {
             // Make PR jobs PR-$num:$cloneJob (not sure how to better handle multiple PR jobs)
-            if (config.trigger === '~pr' && !config.prChain) {
+            if (config.trigger === '~pr') {
                 jobs.add(`PR-${config.prNum}:${edge.dest}`);
             } else {
                 jobs.add(edge.dest);
             }
+        } else if (prChainTrigger && config.trigger === `${prChainTrigger[1]}:${edge.src}`) {
```
Comment (tk3fftk) [inline lib/getNextJobs.js]: I think it's better to add a comment about `prChainTrigger[1]`.

**Pull request merged**